### PR TITLE
Version: 3.32.1

### PR DESCRIPTION
**Pull request by kira-syslogng**
Tarball is available at: https://github.com/syslog-ng/syslog-ng/actions/runs/819321263

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: NEWS.md [hunk 1]
```diff
@@ -1,18 +1,102 @@
...
+   ([#3621](https://github.com/syslog-ng/syslog-ng/pull/3621))
+ * `mongodb()`: template support for the `collection()` option
+   
+   The `collection()` option of the MongoDB destination driver now accepts
+   templates, for example:
+   
+   ```
+   destination {
+     mongodb(
+       uri("mongodb://host/syslog?wtimeoutMS=10000&socketTimeoutMS=10000&connectTimeoutMS=10000&serverSelectionTimeoutMS=5000")
```
Comment (MrAnno) [inline NEWS.md:34]: In the `collection()` example, the URI is unnecessarily long. It was me, sorry.
Could we change the second example something shorter, for example:

```
destination {
  mongodb(
    uri("mongodb://host/syslog")
    collection("${HOST}_messages")
    workers(8)
  );
};
```

File: NEWS.md [hunk 2]
```diff
@@ -1,18 +1,102 @@
...
+ * `csv-parser()`: add drop-invalid() option along with the already existing
+   flag with the same name. This is to improve the consistency of the
+   configuration language.
+   ([#3547](https://github.com/syslog-ng/syslog-ng/pull/3547))
+ * `usertty() destination`: Support changing the terminal disable timeout with the `time-reopen()` option.
+   Default timeout change to 60 from 600. If you wish to use the old 600 timeout, add `time-reopen(600)`
+   to your config in the `usertty()` driver.
+   ([#3585](https://github.com/syslog-ng/syslog-ng/pull/3585))
+ * `syslog-parser()`: add a new drop-invalid() option that allows the use of
+   syslog-parser() in if statements. Normally a syslog-parser() injects an
```
Comment (MrAnno) [inline NEWS.md:59]: This sounds exciting, don't we want to add it to the Highlights section?

(The `collection()` MongoDB option is not such a big deal, we can replace that if you want to.)

File: NEWS.md [hunk 3]
```diff
@@ -1,18 +1,102 @@
...
+   error message instead of failing.
+   ([#3565](https://github.com/syslog-ng/syslog-ng/pull/3565))
+
 ## Bugfixes
 
- * `syslog-parser()`: fix a potential crash in case parsing the message
-   fails and tags are already applied to the message.
+ * date-parser: if the timestamp pattenr did not covered a field (for example seconds) that field had undefined value
```
Comment (MrAnno) [inline NEWS.md:65]: typo: pattenr

File: NEWS.md [hunk 4]
```diff
@@ -1,18 +1,102 @@
...
+   
+   The missing fields are initialized according to the following rules:
+    1) missing all fileds -> use current date
```
Comment (MrAnno) [inline NEWS.md:68]: typo: fileds

File: NEWS.md [hunk 5]
```diff
@@ -1,18 +1,102 @@
...
+    2) only miss year -> guess year based on current year and month (current year, last year or next year)
+    3) the rest of the cases don't make much sense, so zero initialization of the missing field makes sense. And the year is initializeed to the current one.
```
Comment (MrAnno) [inline NEWS.md:70]: typo: initializeed

File: NEWS.md (hunk 2 above)
Comment (gaborznagy) [reply NEWS.md:59]: If this is a vote I would keep the template support in mongodb's `collection` option in the Highlights section. :)

Comment (Kokan) [reply NEWS.md:58]: The `syslog-parser()` new flag is a nice feature, but I do not think a lot of people tinker on that level with the configuration. The template support likely more common for people to have, and it also matches with the other highlight marking this release as a mongodb improvements kinda release. (Sure there are other nice feature it has as well.)

**Review by gaborznagy — COMMENTED** [review]
@MrAnno spotted all the typos... :) Besides that I would recommend #3647 as a packaging note, e.g. `The helper script for preparing grammar generation: lib/merge-grammar.py is changed to use python3 instead of python`

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by Kokan** [comment]
In the last release, we dropped python2 (https://github.com/syslog-ng/syslog-ng/releases/tag/syslog-ng-3.31.2), mentioning that we removed some change that makes python2 less okay is not that helpful.

**Review by MrAnno — DISMISSED** [review]

**Review by gaborznagy — APPROVED** [review]

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**